### DISCUSSION TITLE
Deployment Architecture Discussion: Single ASP.NET Core Project or Multiple APP HOSTS for VM Separation?

### DISCUSSION THREAD

**Discussion by Asimkhan-developer** [microsoft/aspire#976] [General]
Hey Everyone 👋,

I'm currently working on a product that needs to be deployed both stand-alone on the cloud and on my customer's private virtual machines. The customer has provided 4 Virtual Machines, with the requirement that the database should reside on a separate VM, and the application should be deployed on other VMs. Additionally, I'm looking to distribute some services on one VM, some on another, and have the CDN deployed on a separate VM.

Questions:
Single ASP.NET Core Project:

Is it possible to achieve this deployment structure with a single ASP.NET Core project, or would I need to develop different APP HOSTS with connections to different Docker containers present on different VMs?
Deployment Best Practices:

What are the best practices for deploying components of a system on different VMs while ensuring efficient communication and maintainability?
Docker Containerization:

Should I consider containerizing my services and applications to streamline deployment across different VMs?
Database Separation:

Any recommendations on how to best manage the database residing on a separate VM while maintaining smooth interactions with the application?
Looking forward to hearing your insights and experiences! 🚀

**Comment by davidfowl** [comment]
In terms of Aspire's local development, you only need a single app host with your application and the database it references. 

Deploying to VMs is **hard mode** and aspire does nothing to help in this situation. We're optimizing to modern cloud native container-based, orchestrated environments (Azure container apps, ECS, Kubernetes based environments etc). While it would be possible to build something custom to VMs, we have no plans of doing so.
